docs/src/main/asciidoc/spanner-spring-data-r2dbc.adoc

```diff
@@ -0,0 +1,50 @@
+:spring-data-commons-ref: https://docs.spring.io/spring-data/data-commons/docs/current/reference/html
+
+[#spring-data-cloud-spanner-r2dbc]
+== Cloud Spanner Spring Data R2DBC
+
+The Spring Data R2DBC Dialect for Cloud Spanner enables the usage of https://github.com/spring-projects/spring-data-r2dbc[Spring Data R2DBC] with Cloud Spanner.
+
+The goal of the Spring Data project is to create easy and consistent ways of using data access technologies from Spring Framework applications.
+
+=== Setup
+
+Maven coordinates, using <<getting-started.adoc#bill-of-materials, Spring Framework on Google Cloud BOM>>:
+
+[source,xml]
+----
+<dependencies>
+    <!-- The starter dependency for the Spring Data R2DBC framework. -->
+    <dependency>
+        <groupId>org.springframework.boot</groupId>
+        <artifactId>spring-boot-starter-data-r2dbc</artifactId>
+        <version>${r2dbc-version}</version>
```

Comment:
As @burkedavison previous pointed out, we probably want a version here that can be updated as part of the release. e.g. [here](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/pull/1951/files#diff-1194561cbea346acb1bea37d81fdf5c70def982dcbb0c64eca2a96bc51941124L82)
In this case the version here is same as Spring Boot. I wonder if we can reuse pom properties ([this one](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/3.x/pom.xml#L44)) here in adoc?

Reply:
For now I have replaced it with `${spring-boot-dependencies.version}` , but guess it won't be updated automatically as AFAIK a tag like `{x-version-update:spring-boot-dependencies:current}` will update just the POM files (Please let me know if that's not the case)
Do you think we can remove this dependency's version from here and add it to the BOM (`spring-cloud-gcp-dependencies`), it might be simpler for the devs that way as they are anyways importing the BOM
WDYT @burkedavison @meltsufin ?